jerry-core/jerry.c

```diff
@@ -2121,6 +2162,85 @@ jerry_foreach_object_property (const jerry_value_t obj_val, /**< object value */
 } /* jerry_foreach_object_property */
 
 /**
+ * Common part for jerry_resolve_promise and jerry_reject_promise.
+ *
+ * @return undefined value - if success
+ *         value marked with error flag - otherwise
+ */
+static jerry_value_t
+jerry_resolve_or_reject_promise (jerry_value_t promise,
+                                 jerry_value_t argument,
+                                 bool is_resolve)
```

Comment:
My thought is the C api should be similar with js api, so it will be easy to use. In JS world, usually the code is like
``` javascript
var p = new Promise(function(resolve, reject)
{
  ....
  if (some condition)
  {
    var reason = .....//prepare the reason
     reject (reason)
  }
  
  var argument = .... //prepare the argument
   resolve(argument)
})
```

in your example, 
``` c
static void my_get_batt_info_cb (const batt_info_t *info, void *cb_data)
{
        jerry_value_t promise = (jerry_value_t)(uintptr_t) cb_data; 
        
        // do stuff with *info and assign result + should_call_resolve:
        jerry_value_t result = ...;
        bool should_call_resolve = ...;
        
        jerry_call_resolve_reject(promise, result, should_call_resolve);
        jerry_value_release(promise);
}
```
there should be some if-else or branch code in `result = ` and `should_call_resolve =` , e.g.
``` c
if (some condition)
 { result = some_error_reason; should_call_resolve = false; }
else
{result = some_argument; should_call_resolve = true;}
```
why just directly call the resolve/reject in the branch?